src/runner/porter.runner.ts

```diff
@@ -173,6 +175,116 @@ export class PorterRunner extends Runner {
+      member: { data: LastMemberNotesType; errMsg: string } | 'NoError';
+    };
+
+    type ErrMesType = {
+      key: string;
+      date: string;
+      info: ErrInfoType | 'No data found' | 'Data format error';
+    };
+
+    type DataType = {
+      memberNotes: MemberNote;
+      lastMemberNotes: LastMemberNotesType;
+    };
+
+    function hasDataProperty(data: DataType): boolean {
+      return Boolean(
+        data.hasOwnProperty('lastMemberNotes') &&
+          data.lastMemberNotes?.hasOwnProperty('criteria') &&
+          data.lastMemberNotes?.hasOwnProperty('lastMemberRecord') &&
+          data.hasOwnProperty('memberNotes'),
+      );
+    }
+
+    const client = this.cacheService.getClient();
+    const pattern = `PhoneService:*`;
+    const errAry: ErrMesType[] = [];
+    let cursor = '0';
+
+    do {
+      const scanResult = await client.scan(cursor, 'MATCH', pattern, 'COUNT', batchSize);
+      cursor = scanResult[0];
+      const keys = scanResult[1];
+
+      for (const key of keys) {
+        const valueString = await client.get(key);
+        const dateTime = key.split(':')[1];
+        const errMesInit: ErrMesType = {
+          key,
+          date: `${new Date(parseInt(dateTime, 10))}`,
+          info: {
+            memberNote: 'NoError',
+            member: 'NoError',
+          },
+        };
+
+        if (!valueString) {
+          errMesInit.info = 'No data found';
+          errAry.push(errMesInit);
+          continue;
+        }
+        const data: DataType = JSON.parse(valueString);
+
+        if (!hasDataProperty(data)) {
+          errMesInit.info = 'Data format error';
+          errAry.push(errMesInit);
+          continue;
+        }
+        const { memberNotes, lastMemberNotes } = data;
+        const { criteria, lastMemberRecord } = lastMemberNotes;
+
+        try {
+          await manager.getRepository(MemberNote).insert(memberNotes);
+        } catch (error) {
+          if (typeof errMesInit.info !== 'string' && errMesInit.info.memberNote === 'NoError') {
+            errMesInit.info.memberNote = { data: memberNotes, errMsg: error.toString() };
+          }
+        }
+        try {
+          await manager.getRepository(Member).update(criteria, lastMemberRecord);
+        } catch (error) {
+          if (typeof errMesInit.info !== 'string' && errMesInit.info.member === 'NoError') {
+            errMesInit.info.member = { data: lastMemberNotes, errMsg: error.toString() };
+          }
+        }
+
+        if (
+          typeof errMesInit.info !== 'string' &&
+          (typeof errMesInit.info.member !== 'string' || typeof errMesInit.info.memberNote !== 'string')
+        ) {
+          errAry.push(errMesInit);
+        }
+
+        await client.del(key);
+      }
+    } while (cursor !== '0');
+
+    if (errAry.length > 0) {
+      for (const item of errAry) {
+        console.error(`Saving phone service failed:${item}`);
+        console.log(item);
```

Comment:
looks like we have an extra console.log(item) here

Reply:
NITS: we can add this config into our linting file
https://stackoverflow.com/a/49990579